### PR TITLE
Libmongocrypt binding docstrings

### PR DESCRIPTION
**Pull request by egiurleo**

### CODE REVIEW

**Comment by egiurleo** [comment]
I realize now that this is going to create a big merge conflict with the refactoring PR. I'll hold off on merging and then fix it once the other PR is merged.

**Comment by egiurleo** [comment]
@p-mongo I updated this PR for the latest master and added docstrings to the new helper methods.

File: lib/mongo/crypt/binding.rb [hunk 1]
```diff
@@ -130,31 +182,71 @@ class Binding
...
+        :bool
+      )
 
+      # Set the logger callback function on the Mongo::Crypt::Handle object
+      #
+      # @param [ Mongo::Crypt::Handle ] handle
+      # @param [ Method ] log_callback
+      #
+      # @raise [ Mongo::CryptError ] If the callback is not set successfully
       def setopt_log_handler(handle, log_callback)
```
Comment (p-mongo) [inline lib/mongo/crypt/binding.rb]: This seems to be missing `self.` so it's not invoked anywhere?

Comment (p-mongo) [reply lib/mongo/crypt/binding.rb]: Or only on your local test runs with logging on.

File: lib/mongo/crypt/binding.rb [hunk 2]
```diff
@@ -306,11 +464,52 @@ def self.ctx_encrypt_init(context, db_name, an_int, command)
...
+        [:pointer, :pointer],
+        :bool
+      )
+
+      # Initialize the Context for explicit encryption
+      #
+      # @param [ Mongo::Crypt::Context ] context
+      #
+      # @raise [ Error::CryptError ] If initialization fails
+      def self.ctx_explicit_encrypt_init(context, doc)
```
Comment (p-mongo) [inline lib/mongo/crypt/binding.rb]: The `doc` does not appear to be documented.

File: lib/mongo/crypt/binding.rb [hunk 3]
```diff
@@ -306,11 +464,52 @@ def self.ctx_encrypt_init(context, db_name, an_int, command)
...
+      #
+      # @return [ Boolean ] Whether the initialization was successful
       attach_function :mongocrypt_ctx_decrypt_init, [:pointer, :pointer], :bool
 
+      # Initialize the Context for auto-decryption
+      #
+      # @param [ Mongo::Crypt::Context ] context
+      #
+      # @raise [ Error::CryptError ] If initialization fails
       def self.ctx_decrypt_init(context, command)
```
Comment (p-mongo) [inline lib/mongo/crypt/binding.rb]: `command` doc missing

File: lib/mongo/crypt/binding.rb [hunk 4]
```diff
@@ -320,11 +519,34 @@ def self.ctx_decrypt_init(context, command)
...
+        [:pointer, :pointer],
+        :bool
+      )
+
+      # Initialize the Context for explicit decryption
+      #
+      # @param [ Mongo::Crypt::Context ] context
+      #
+      # @raise [ Error::CryptError ] If initialization fails
+      def self.ctx_explicit_decrypt_init(context, doc)
```
Comment (p-mongo) [inline lib/mongo/crypt/binding.rb]: `doc` documentation missing

File: lib/mongo/crypt/binding.rb (hunk 1 above)
Comment (egiurleo) [reply lib/mongo/crypt/binding.rb]: Yeah you're right only the local tests have logging. Adding `self`.

**Pull request merged**